File: Spring 23/drawcheck/templates/drawcheck_template.py
def solve() -> None:
    """
    Find the number on the judge's card through draw and check queries while
    minimizing the average number of queries needed.
    
    Call the draw and check functions below to make draw and check queries. Return
    from this function when you see check() return ABSENT. This template also
    handles reading the final verdict at the end.
    """
    # YOUR CODE HERE


def draw() -> list[int]:
    """
    Shuffle the deck and draw the top 17 cards.
    """
    print('draw', flush=True)
    response = input()
    if response == 'WRONG_ANSWER':
        exit()
    return list(map(int, response.split()))


def check(card: int) -> str:
    """
    Check if a card exists within the deck.
    """
    print('check', card, flush=True)
    response = input()
    if response == 'WRONG_ANSWER':
        exit()
    return response


def main():
    T = int(input())
    for _ in range(T):
        solve()
    verdict = input()


if __name__ == '__main__':
    main()

File: Spring 23/drawcheck/templates/drawcheck_template.cpp
#include <cassert>
#include <iostream>
#include <vector>

using namespace std;

vector<int> draw();
string check(int);

/*
 * Find the number on the judge's card through draw and check queries while
 * minimizing the average number of queries needed.
 * 
 * Call the draw and check functions below to make draw and check queries. Return
 * from this function when you see check() return ABSENT. This template also
 * handles reading the final verdict at the end.
*/
void solve() {
    // YOUR CODE HERE
}

/*
 * Shuffle the deck and draw the top 17 cards.
 */
vector<int> draw() {
    cout << "draw" << endl;
    int K = 17;
    vector<int> cards(K);
    for (int &x: cards) {
        cin >> x;
    }
    return cards;
}

/*
 * Check if a card exists within the deck.
 */
string check(int card) {
    cout << "check " << card << endl;
    string result;
    cin >> result;
    return result;
}

int main() {
    int T;
    cin >> T;
    for (int i = 0; i < T; i++) {
        solve();
    }
    string temp;
    cin >> temp;
}
